source/tutorials/ad-hoc-developer-environments.rst

```diff
@@ -0,0 +1,137 @@
+Ad hoc developer environments
+=============================
+
+Assuming you have :ref:`Nix installed <install-nix>`, it is able to download packages
+and provide a new shell with packages available.
```

Comment:
“**shell environment**” is the main idea being discussed. Be sure to highlight that in bold and re-use this name prominently everywhere else.

Reply:
:+1: